### PR TITLE
allow json library to be configurable, run `mix format`

### PR DESCRIPTION
**Pull request by mus0u**
howdy folks! thanks so much for this library. it's been working like a charm for me so far. i really appreciate y'all putting it out there for randos like myself.

one thing i did wish for was the ability to configure what JSON codec library was being used to decode the user info. most of my projects use [Jason](https://github.com/michalmuskala/jason) for this, so this was the only library that was pulling in Poison into my deps. the interface is basically the same between the two libraries, and other projects, most notably Phoenix, have made the choice of JSON library configurable. i've chosen to emulate the name of Phoenix's configuration setting in this PR, but i don't really have any preference on the name. it just seemed like a reasonable choice.

i also ran `mix format` at the root of the project, because i noticed that my Emacs was autoformatting some things on save, and i wanted everything to be uniform. if you'd prefer a version of this PR without any formatting changes, i completely understand (i have my own beef with a few of mix format's aesthetic choices...), and i'm happy to submit an alternate version that just adds the config changes, if that'd be preferable to y'all.

anyway, if this PR is entirely unwelcome, then i apologize and please close it without a second thought! i just thought i'd offer it up, since it made my life a little bit easier. if it is welcome, then i hope i've made it easy to integrate. i have some ideas for other small features i might want to offer, such as facilities for stubbing the HTTP call in test mode with Mox, if it's amenable to y'all.

thanks for reading, and have a great week!

### CODE REVIEW

File: lib/config.ex [hunk 1]
```diff
@@ -24,4 +26,6 @@ defmodule Auth0Jwks.Config do
   def iss, do: get_config_or_error(:auth0_jwks, :iss)
   def aud, do: get_config_or_error(:auth0_jwks, :aud)
   def jwks_url, do: iss() <> ".well-known/jwks.json"
+
+  def json_library, do: Application.get_env(:auth0_jwks, :json_library) || Poison
```
Comment (mus0u) [inline lib/config.ex:30]: here is the first non-`mix format`-based change in this PR...

File: lib/user_info.ex [hunk 2]
```diff
@@ -1,12 +1,15 @@
 defmodule Auth0Jwks.UserInfo do
   use HTTPoison.Base
 
+  alias Auth0Jwks.Config
```
Comment (mus0u) [inline lib/user_info.ex:4]: here is the second non-`mix format`-based change in this PR...

File: lib/user_info.ex [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
+
   def from_token(token) do
     auth_header = {"Authorization", "Bearer #{token}"}
+
     with {:ok, response} <- get("userinfo", [auth_header]) do
       response
       |> Map.get(:body)
-      |> Poison.decode()
+      |> Config.json_library().decode()
```
Comment (mus0u) [inline lib/user_info.ex:12]: ...and here's the third one. there's just one more!

File: mix.exs [hunk 4]
```diff
@@ -36,7 +35,7 @@ defmodule Auth0Jwks.MixProject do
       {:joken, "~> 2.0"},
       {:joken_jwks, "~> 1.1.0"},
       {:plug_cowboy, "~> 2.0"},
-      {:poison, "~> 4.0"},
+      {:poison, "~> 4.0", optional: true}
```
Comment (mus0u) [inline mix.exs:38]: this is the last one. everything else was just `mix format`. hopefully this makes it easy to read.

File: lib/token.ex [hunk 5]
```diff
@@ -6,13 +6,16 @@ defmodule Auth0Jwks.Token do
   @impl true
   def token_config do
     default_claims(skip: [:aud, :iss])
-    |> add_claim("iss", nil, &issued_by_auth0_domain?/1) # todo create util
+    # todo create util
```
Comment (bnchrch) [inline lib/token.ex:9]: Could we remove this todo?

**Review by bnchrch — CHANGES_REQUESTED** [review]
@mus0u Hey thanks for the PR! So sorry for the late review I just saw this!

Anyway this seems fine to merge, if you'd be so kind to remove the todo and just update the readme to include the json_library feature!

**Pull request merged**